src/components/Checkbox/Checkbox.tsx

```diff
@@ -314,6 +314,9 @@ export type CheckboxProps = BaseFormElementWithNodeLabel &
     indeterminate?: boolean;
   };
 
+/**
+ * A checkbox component for form usage. Under the covers, this is an input element with `type="checkbox"`.
```

Comment:
Instead of using mdx we can just add comments in the code and it will pick them out and add it to the storybook. In general, I think is self explanatory and we don't need as much text.